src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -63,7 +63,7 @@ public void AddOrUpdate(FilePath file, JObject? metadata, string? outputPath)
                     SourceFile = sourceFile.Origin == FileOrigin.Redirection ? null : sourceFile,
                     SourcePath = sourceFile.Origin == FileOrigin.Redirection ? null : sourceFilePath,
                     SourceUrl = _contributionProvider.GetReportGitUrl(sourceFile),
-                    Locale = _locale,
+                    Locale = buildOutput.metadata!.ContainsKey("locale") ? null : _locale,
```

Comment:
The `buildOutput.metadata` maybe null in some case, use `!` will cause exception
```suggestion
                    Locale = buildOutput.metadata?.ContainsKey("locale") ? null : _locale,
```